modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/to/PodcastGrouping.kt

```diff
@@ -9,21 +9,35 @@ import au.com.shiftyjelly.pocketcasts.models.to.PodcastGrouping.Season.getSeason
 import au.com.shiftyjelly.pocketcasts.models.type.EpisodesSortType
 import au.com.shiftyjelly.pocketcasts.localization.R as LR
 
-sealed class PodcastGrouping(@StringRes val groupName: Int, val sortFunction: ((PodcastEpisode) -> Int)?) {
+sealed class PodcastGrouping(
+    @StringRes val groupName: Int,
+    val serverId: Int,
+    val sortFunction: ((PodcastEpisode) -> Int)?,
+) {
     companion object {
         val All
             get() = listOf(None, Downloaded, Unplayed, Season, Starred)
+
+        fun fromServerId(id: Int) = All.find { it.serverId == id } ?: None
```

Comment:
We should replace `find` with `firstOrNull` for reducing the unneccesary iterations.

Reply:
What unnecessary iterations?

```kotlin
@kotlin.internal.InlineOnly
public inline fun <T> Iterable<T>.find(predicate: (T) -> Boolean): T? {
    return firstOrNull(predicate)
}
```

https://github.com/JetBrains/kotlin/blob/09f2f365452396a0b6b835b1b7862c5d1b33dc69/libraries/stdlib/common/src/generated/_Collections.kt#L165-L168